impeller/entity/geometry/superellipse_geometry.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <vector>
+
+#include "flutter/impeller/entity/geometry/superellipse_geometry.h"
+
+#include "impeller/geometry/constants.h"
+
+namespace impeller {
+
+SuperellipseGeometry::SuperellipseGeometry(const Point& center,
+                                           Scalar radius,
+                                           Scalar degree,
+                                           Scalar alpha,
+                                           Scalar beta)
+    : center_(center),
+      degree_(degree),
+      radius_(radius),
+      alpha_(alpha),
+      beta_(beta) {}
+
+GeometryResult SuperellipseGeometry::GetPositionBuffer(
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass) const {
+  // https://math.stackexchange.com/questions/2573746/superellipse-parametric-equation
+  Scalar a = alpha_;
+  Scalar b = beta_;
+  Scalar n = degree_;
+
+  // TODO(jonahwilliams): determine parameter values based on scaling factor.
+  Scalar step = kPi / 80;
+
+  // Generate the points for the top left quadrant, and then mirror to the other
+  // quadrants.
+  std::vector<Point> points;
+  points.reserve(41);
+  for (int i = 0; i <= 40; i++) {
+    Scalar t = i * step;
+    Scalar x = a * pow(abs(cos(t)), 2 / n);
+    Scalar y = b * pow(abs(sin(t)), 2 / n);
+    points.emplace_back(x, y);
+  }
+
+  static constexpr Point reflection[4] = {{1, 1}, {-1, 1}, {-1, -1}, {1, -1}};
+
+  // Reflect into the 4 quadrants and generate the tessellated mesh. The
+  // iteration order is reversed so that the trianges are continuous from
+  // quadrant to quadrant.
+  std::vector<Point> geometry;
+  geometry.reserve(1 + 4 * points.size());
+  geometry.push_back(center_);
+  for (auto i = 0u; i < points.size(); i++) {
+    geometry.push_back(center_ + ((reflection[0] * points[i]) * radius_));
```

Comment:
Does OOO matter for these equations? ((ref * pt) * rad) vs (ref * (pt * rad)) vs just (ref * pt * rad)?

Reply:
(Relates to my comment above about just multiplying by radius in the points array...)

Reply:
No, removed